modules/apps/asset/asset-auto-tagger-service/src/test/java/com/liferay/asset/auto/tagger/internal/configuration/persistence/listener/AssetAutoTaggerCompanyConfigurationModelListenerTest.java

```diff
@@ -0,0 +1,138 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.asset.auto.tagger.internal.configuration.persistence.listener;
+
+import com.liferay.asset.auto.tagger.configuration.AssetAutoTaggerConfiguration;
+import com.liferay.asset.auto.tagger.configuration.AssetAutoTaggerConfigurationFactory;
+import com.liferay.portal.configuration.persistence.listener.ConfigurationModelListenerException;
+import com.liferay.portal.kernel.test.ReflectionTestUtil;
+import com.liferay.portal.kernel.test.util.RandomTestUtil;
+import com.liferay.portal.kernel.util.HashMapDictionary;
+import com.liferay.portal.kernel.util.ResourceBundleUtil;
+
+import java.util.Dictionary;
+import java.util.Locale;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.mockito.Matchers;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+/**
+ * @author Alicia Garcia
+ */
+@PrepareForTest(ResourceBundleUtil.class)
+@RunWith(PowerMockRunner.class)
+public class AssetAutoTaggerCompanyConfigurationModelListenerTest {
+
+	@Before
+	public void setUp() {
+		MockitoAnnotations.initMocks(this);
+		_setUpAssetAutoTaggerCompanyConfigurationModelListener();
+		_setUpResourceBundleUtil();
+	}
+
+	@Test(expected = ConfigurationModelListenerException.class)
+	public void testMaximumNumberOfTagsPerAssetGreaterThanSystem()
+		throws ConfigurationModelListenerException {
+
+		_setUpAssetAutoTaggerConfigurationFactory();
+
+		ReflectionTestUtil.setFieldValue(
```

Comment:
We need to make sure we undo any side effects to make sure any other tests are not affected.